packages/app/src/app/(consultation)/_representation-equilibree/recherche/FormSearchSiren.tsx

```diff
@@ -0,0 +1,165 @@
+"use client";
+
+import { ButtonsGroup } from "@codegouvfr/react-dsfr/ButtonsGroup";
+import { Input } from "@codegouvfr/react-dsfr/Input";
+import { Select } from "@codegouvfr/react-dsfr/Select";
+import {
+  type COUNTIES,
+  FULL_SORTED_REGIONS_TO_COUNTIES,
+  type NAF_SECTIONS,
+  type REGIONS,
+  SORTED_NAF_SECTIONS,
+  SORTED_REGIONS,
+} from "@common/dict";
+import { omitByRecursively } from "@common/utils/object";
+import { capitalize } from "@common/utils/string";
+import { Grid, GridCol } from "@design-system";
+import _ from "lodash";
+import { useRouter } from "next/navigation";
+import { useCallback, useEffect } from "react";
+import { useForm } from "react-hook-form";
+
+export type FormTypeInput = {
+  departement?: keyof typeof COUNTIES;
+  naf?: keyof typeof NAF_SECTIONS;
+  q?: string;
+  region?: keyof typeof REGIONS;
+};
+
+export interface FormSearchSirenProps {
+  searchParams: FormTypeInput;
+}
+
+export const FormSearchSiren = ({ searchParams }: FormSearchSirenProps) => {
+  const router = useRouter();
+
+  const {
+    formState: { errors },
+    handleSubmit,
+    register,
+    reset,
+    watch,
+    // setValue,
+  } = useForm<FormTypeInput>(); // Using defaultValues would not be enough here, because we fetch params.query which is not present at the first rehydration time (see Next.js docs). So we use reset API below.
+
+  const resetInputs = useCallback(
+    (params: FormTypeInput) => {
+      reset({
+        region: params.region,
+        departement: params.departement,
+        naf: params.naf,
+        q: params.q,
+      });
+    },
+    [reset],
+  );
+
+  // Sync form data with URL params.
+  useEffect(() => {
+    resetInputs(searchParams);
+  }, [resetInputs, searchParams]);
+
+  const regionSelected = watch("region");
+
+  function onSubmit(data: FormTypeInput) {
+    router.replace(
+      `${location.pathname}?${new URLSearchParams(
+        omitByRecursively(data, _.isUndefined) as Partial<FormTypeInput>,
```

Comment:
C'est mieux, je pense, d'utiliser directement isUndefined de lodash, plutôt qu'importer _, afin de ne pas alourdir la page.

Ex de la doc: 
// Cherry-pick methods for smaller browserify/rollup/webpack bundles.
var at = require('lodash/at');